### PR TITLE
wip graalvm build

### PR DESCRIPTION
**Pull request by Mic92**
fixes https://github.com/marytts/marytts/issues/1000
if it works...

### CODE REVIEW

**Comment by Mic92** [comment]
Currently fails with: 

```
ash-4.2# ./gradlew nativeImage

> Task :nativeImage
Build on Server(pid: 231, port: 35909)
[hello-world:231]    classlist:   4,837.69 ms,  2.54 GB
[hello-world:231]        (cap):     306.68 ms,  2.54 GB
[hello-world:231]        setup:     584.99 ms,  2.54 GB
[hello-world:231]     analysis:   5,972.58 ms,  2.54 GB
Error: Classes that should be initialized at run time got initialized during image building:
 org.hsqldb.jdbc.JDBCDriver was unintentionally initialized at build time. To see why org.hsqldb.jdbc.JDBCDriver got initialized use -H:+TraceClassInitialization

Error: Use -H:+ReportExceptionStackTraces to print stacktrace of underlying exception
Error: Image build request failed with exit status 1

> Task :nativeImage FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':nativeImage'.
> Process 'command '/root/.gradle/caches/com.palantir.graal/20.0.0/8/graalvm-ce-java8-20.0.0/bin/native-image'' finished with non-zero exit value 1

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.3/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 13s
77 actionable tasks: 1 executed, 76 up-to-date
```

If someone has a suggestions, feel free to throw it it.

**Comment by Mic92** [comment]
One can also get this Dockefile to get graalvm quickly:

```
FROM oracle/graalvm-ce:19.0.0
WORKDIR /opt/graalvm
RUN gu install native-image

ENTRYPOINT ["bash"]
```

**Comment by Mic92** [comment]
This thread might provide some insights: https://github.com/oracle/graal/issues/966

File: marytts-runtime/src/main/java/marytts/util/MaryCache.java [hunk 1]
```diff
@@ -94,31 +95,31 @@ public static boolean haveCache() {
 	 */
 	public MaryCache(File cacheFile, boolean clearCache) throws ClassNotFoundException, SQLException {
 		// Load the HSQL Database Engine JDBC driver
-		Class.forName("org.hsqldb.jdbcDriver");
```
Comment (Mic92) [inline marytts-runtime/src/main/java/marytts/util/MaryCache.java:97]: Seems like this breaks the build.

**Comment by Mic92** [comment]
It builds now, but it seems that the native-image version needs actually more memory. It currently needs to embed a jdk, I assume for groovy...

**Comment by psibre** [comment]
IIRC there's a Java property one can set to disable the cache -- although I'm not sure if it would disable all of the JDBC stuff that seems to fail here.

Interesting to see if it's just this, or if other reflection-based service loading is also a problem.

**Comment by psibre** [comment]
> embed a jdk, I assume for groovy

Yes, we want to get rid of Groovy leaking into the API, but this is currently still on our TODO list...

**Comment by Mic92** [comment]
Ok. I keep this on hold until you can get rid of groovy.

**Comment by Mic92** [comment]
> IIRC there's a Java property one can set to disable the cache -- although I'm not sure if it would disable all of the JDBC stuff that seems to fail here.
> 
> Interesting to see if it's just this, or if other reflection-based service loading is also a problem.

It was just this. when I commented it out, it the build succeeded.

**Review by mary-doc — APPROVED** [review]
@

**Comment by Mic92** [comment]
I switched to mozilla's TTS...

**Pull request closed without merging**